File: Android/tuilivekit/livekit/src/main/java/com/trtc/uikit/livekit/component/gift/view/animation/manager/AnimationPlayer.java
package com.trtc.uikit.livekit.component.gift.view.animation.manager;

public abstract class AnimationPlayer {
    public abstract void preparePlay(GiftAnimationModel model);

    public abstract void startPlay(GiftAnimationModel model);

    public abstract void stopPlay();

    public abstract void setCallback(PlayCallback callback);

    public interface PlayCallback {
        void onFinished(int error);
    }
}

File: Android/tuilivekit/src/main/java/com/trtc/uikit/livekit/component/gift/view/animation/manager/AnimationPlayer.kt
package com.trtc.uikit.livekit.component.gift.view.animation.manager

import com.trtc.uikit.livekit.component.gift.viewmodel.GiftModel

abstract class AnimationPlayer {
    abstract fun preparePlay(model: GiftModel)

    abstract fun startPlay(model: GiftModel)

    abstract fun stopPlay()

    abstract fun setCallback(callback: PlayCallback?)

    fun interface PlayCallback {
        fun onFinished(error: Int)
    }
}